### PR TITLE
Add migrate_pmid_to_uid tool (for Seagull metadata migration)

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate This script migrates the Seagull metadata database from using the `_id` (private.meta.id) to `userId` as its primary key. It would run before (and after to ensure everything was converted) deploying the next version of Seagull.

### CODE REVIEW

File: tools/migrate_pmid_to_uid/migrate_pmid_to_uid.go [hunk 1]
```diff
@@ -0,0 +1,366 @@
+package main
+
+/* CHECKLIST
```
Comment (jh-bate) [inline tools/migrate_pmid_to_uid/migrate_pmid_to_uid.go]: do we need to update the checklist or is it not relevant?

File: tools/migrate_pmid_to_uid/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+## migrate_pmid_to_uid
```
Comment (jh-bate) [inline tools/migrate_pmid_to_uid/README.md]: do you think we should add the reason or link to the details as to why we are doing this? Or is it a case of once its done it no longer matters?

Comment (darinkrauss) [inline tools/migrate_pmid_to_uid/README.md]: It is a "done once, never again" migration.

However, what we really need is some sort of overall changelog that would detail these things for anyone with their own setup. Not specific per repo, I guess, as that would most likely not be seen. Basically, some reliable way to communicate with anyone using the code.

File: tools/migrate_pmid_to_uid/migrate_pmid_to_uid.go (hunk 1 above)
Comment (darinkrauss) [inline tools/migrate_pmid_to_uid/migrate_pmid_to_uid.go]: I'll leave it as-is. Probably doesn't belong in this file, but all of the files probably should be rev'd to remove the checklist.

Comment (jh-bate) [inline tools/migrate_pmid_to_uid/migrate_pmid_to_uid.go]: ok

File: tools/migrate_pmid_to_uid/README.md (hunk 2 above)
Comment (jh-bate) [inline tools/migrate_pmid_to_uid/README.md]: true. ok i guess that com's method goes on the platform backlog then :)

**Comment by darinkrauss** [comment]
[[backend] Remove metadata dependency upon user private meta id](https://trello.com/c/7rsj25Mi/223-backend-remove-metadata-dependency-upon-user-private-meta-id)

**Pull request merged**